src/main/java/com/laborete/LaboreteAPI/profile/services/UsersServiceImpl.java

```diff
@@ -145,9 +261,24 @@ private String createDirectory(UUID uuid) {
                 Files.createDirectory(thirdSymbolDirectory.toPath());
             }
 
-            return ROOT_PATH + firstSymbol + "\\" + secondSymbol + "\\" + thirdSymbol;
+            return parentDirectory + firstSymbol + "\\" + secondSymbol + "\\" + thirdSymbol;
         } catch (Exception e) {
-            throw new ResourceFileUploadErrorException(e.getMessage());
+            throw new ResourceFileUploadErrorException(ERROR_CREATING_DIRECTORY + e.getMessage());
         }
+
     }
+
+    private String getDirectory(UUID uuid, String parentDirectory) {
```

Comment:
Consider renaming `uuid` param - what does it mean? Is it uuid of a user, or is it uuid of image?